### PR TITLE
better cython debugging

### PR DESCRIPTION
**Pull request by WillAyd**
This is still super buggy but hopefully this gets people a bit closer

### CODE REVIEW

File: doc/source/development/debugging_extensions.rst [hunk 1]
```diff
@@ -14,14 +14,31 @@ For Python developers with limited or no C/C++ experience this can seem a daunti
...
+
+   docker run --rm -it -w /data -v ${PWD}:/data pandas/pandas-debug
+
+Inside the image, you can use meson to build/install pandas and place the build artifacts into a ``debug`` folder using a command as follows:
+
+.. code-block:: sh
+
+    python -m pip install -ve . --no-build-isolation --config-settings=builddir="debug" --config-settings=setup-args="-Dbuildtype=debug"
+
+If planning to use cygdb, the files required by that application are placed within the build folder. So you have to first ``cd`` to the build folder, then start that application.
```
Comment (WillAyd) [inline doc/source/development/debugging_extensions.rst:34]: I think this is a bug in Cython, which still uses optparse (deprecated in 3.2) insted of argparse. Seems like the tool should allow you to specify the build folder in theory, but in practice doesn't work. Will push up a PR separately to Cython to fix

Comment (WillAyd) [reply doc/source/development/debugging_extensions.rst:34]: See https://github.com/cython/cython/pull/5499

**Review by MarcoGorelli — APPROVED** [review]
Nice! I haven't tried the docker image yet, but the changes suggested look good

**Comment by MarcoGorelli** [comment]
looks like some merge conflicts have crept in - if they're resolved, then can this be merged?

**Comment by datapythonista** [comment]
@WillAyd is this something we want to move forward with? Can you fix the conflicts if it is please.

**Comment by WillAyd** [comment]
Sorry forgot about this one. Yea I think its only helpful so happy to move forward with it

**Comment by datapythonista** [comment]
Thanks for this @WillAyd

**Pull request merged**